events/ri/README.md

```diff
@@ -0,0 +1,137 @@
+
+[Apache Kafka]:https://kafka.apache.org/
+
+### Configuration
+
+The `KafkaEventSubscriber` implementation reads its configuration from a properties file. The default location of this
+file is `./nessie-kafka.properties`, but this can be changed by setting the `nessie.events.config.file` system property
+or the `NESSIE_EVENTS_CONFIG_FILE` environment variable to a different location.
+
+The properties file should contain all the desired Kafka producer configuration options, as well as the following
+subscriber-specific options:
+
+- `nessie.events.topic` - the name of the Kafka topic to publish events to. Defaults to `nessie-events`.
+- `nessie.events.repository-ids` - a comma-separated list of repository ids to watch. If not set, all repositories will
+  be watched. See below for more details.
+
+A template configuration file can be found in `src/main/resources/nessie-kafka.template.properties`.
+
+### Serialization and deserialization
+
+`KafkaEventSubscriber` uses the [Avro] library to define a domain model that is specific to the subscriber, and is not
+tied in any way to Nessie's. This is indeed the recommended approach, as it allows the subscriber to define its own
+domain model and schema, and to evolve them independently of Nessie.
+
+[Avro]:https://avro.apache.org/
+
+The domain model is defined in the `src/main/avro/NessieEvents.avdl` file. The Avro compiler and its Gradle plugin are
+used during build to generate Java classes from the Avro schema files. The generated classes are located in the
+`build/generated-main-avro-java` directory.
+
+Note: for the consumer to be able to deserialize the messages into the same generated classes, it must be configured
+with `specific.avro.reader=true`.
+
+### Schema and polymorphism
+
+Nessie's Events API is polymorphic, in the sense that it supports different types of events. The `KafkaEventSubscriber`
+implementation illustrates how to handle different types of events in the same subscriber, pushing to a single Kafka
+topic. 
+
+In this case, the subscriber aggregates Nessie's event types into three kinds of Avro messages: `CommitEvent`,
+`ReferenceEvent`, and `OperationEvent`.
+
+- `CommitEvent` is generated when a commit is made to a branch and corresponds to Nessie's `COMMIT` event type;
+- `ReferenceEvent` is generated when a branch or tag is created, updated, or deleted, and corresponds to Nessie's
+  `REFERENCE_CREATED`, `REFERENCE_UPDATED`, and `REFERENCE_DELETED` event types;
+- `OperationEvent` is generated when a PUT or a DELETE operation occurs, and corresponds to Nessie's
+  `CONTENT_STORED` and `CONTENT_REMOVED` event types.
+
+In this example, the destination Kafka topic is going to receive messages of three different types. This is a common
+scenario in Kafka, but requires some extra work to be done on both the producer and consumer sides, especially with
+respect to schema handling. The strategy we chose to use here is outlined in [Martin Kleppmann's excellent article] and
```

Comment:
Can you elaborate more on the most important aspect in our context of why we picked this strategy? 
Was it `any events that need to stay in a fixed order must go in the same topic`

Reply:
I chose this strategy for the reference impl because this strategy does not require to pre-configure the schema registry, which makes it easier to test the producer. 

IOW, when the topic is created and the first message comes in, the producer will upload the schema since it's not present in the registry yet. It will do this for each new message type, since the strategy is to have one schema per (topic + message type) pair. 

Other strategies, for example those outlined in Robert Yokota's article, require the registry to be set up will all the schemas prior to using the topic.